sqlmesh/core/model/definition.py

```diff
@@ -628,6 +636,104 @@ def is_breaking_change(self, previous: Model) -> t.Optional[bool]:
         """
         raise NotImplementedError
 
+    @property
+    def data_hash(self) -> str:
+        """
+        Computes the data hash for the node.
+
+        Returns:
+            The data hash for the node.
+        """
+        data = [
+            str(self.sorted_python_env),
+            self.kind.name,
+            self.cron,
+            self.storage_format,
+            str(self.lookback),
+            *(expr.sql() for expr in (self.partitioned_by or [])),
+            *(self.clustered_by or []),
+            self.stamp,
+            *self._additional_data_for_hash,
+        ]
+
+        for column_name, column_type in (self.columns_to_types_ or {}).items():
+            data.append(column_name)
+            data.append(column_type.sql())
+
+        if isinstance(self.kind, IncrementalByTimeRangeKind):
+            data.append(self.kind.time_column.column)
+            data.append(self.kind.time_column.format)
+        elif isinstance(self.kind, IncrementalByUniqueKeyKind):
+            data.extend(self.kind.unique_key)
+
+        return hash_data(data)
+
+    @property
+    def _additional_data_for_hash(self) -> t.List[str]:
```

Comment:
[Nit] I guess the name doesn't make it clear that this is for data hash specifically

Reply:
Refactored